subsys/fs/fat_fs.c

```diff
@@ -516,7 +516,7 @@ static const struct fs_file_system_t fatfs_fs = {
 	.mkdir = fatfs_mkdir,
 	.stat = fatfs_stat,
 	.statvfs = fatfs_statvfs,
-#if defined(CONFIG_FILE_SYSTEM_MKFS)
+#if defined(CONFIG_FILE_SYSTEM_MKFS) && defined(CONFIG_FS_FATFS_MKFS)
```

Comment:
Since ``CONFIG_FS_FATFS_MKFS` depends on `CONFIG_FILE_SYSTEM_MKFS` it should be enough to chek the first one only.